sim/shaman/chain_lightning.go

```diff
@@ -8,38 +8,50 @@ import (
 )
 
 func (shaman *Shaman) registerChainLightningSpell() {
-	numHits := min(core.TernaryInt32(shaman.HasMajorGlyph(proto.ShamanMajorGlyph_GlyphOfChainLightning), 4, 3), shaman.Env.GetNumTargets())
+	numHits := min(core.TernaryInt32(shaman.HasMajorGlyph(proto.ShamanMajorGlyph_GlyphOfChainLightning), 5, 3), shaman.Env.GetNumTargets())
 	shaman.ChainLightning = shaman.newChainLightningSpell(false)
-	shaman.ChainLightningLOs = []*core.Spell{}
+	shaman.ChainLightningOverloads = []*core.Spell{}
 	for i := int32(0); i < numHits; i++ {
-		shaman.ChainLightningLOs = append(shaman.ChainLightningLOs, shaman.newChainLightningSpell(true))
+		shaman.ChainLightningOverloads = append(shaman.ChainLightningOverloads, shaman.newChainLightningSpell(true))
 	}
 }
 
-func (shaman *Shaman) newChainLightningSpell(isLightningOverload bool) *core.Spell {
+func (shaman *Shaman) newChainLightningSpell(isElementalOverload bool) *core.Spell {
+	castTime := time.Millisecond * 2500
+	spellCoeff := 0.571
+	canOverload := false
+	overloadChance := shaman.GetOverloadChance()
+	if shaman.Spec == proto.Spec_SpecElementalShaman {
+		castTime -= 500
+		// 0.36 is shamanism bonus
+		spellCoeff += 0.36
+		canOverload = true
+	}
+
 	spellConfig := shaman.newElectricSpellConfig(
-		core.ActionID{SpellID: 49271},
+		core.ActionID{SpellID: 421},
 		0.26,
-		time.Millisecond*2000,
-		isLightningOverload)
+		castTime,
+		isElementalOverload)
 
 	numHits := min(core.TernaryInt32(shaman.HasMajorGlyph(proto.ShamanMajorGlyph_GlyphOfChainLightning), 4, 3), shaman.Env.GetNumTargets())
-	dmgReductionPerBounce := core.TernaryFloat64(shaman.HasSetBonus(ItemSetTidefury, 2), 0.83, 0.7)
-	spellCoeff := 0.5714 + 0.04*float64(shaman.Talents.Shamanism)
+	dmgReductionPerBounce := 0.7
+	baseDamage := 1093.0
 
-	canLO := !isLightningOverload && shaman.Talents.LightningOverload > 0
-	lightningOverloadChance := float64(shaman.Talents.LightningOverload) * 0.11 / 3
+	if shaman.HasMajorGlyph(proto.ShamanMajorGlyph_GlyphOfChainLightning) {
+		baseDamage *= 0.90
+	}
 
 	spellConfig.ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
 		bounceCoeff := 1.0
 		curTarget := target
 		for hitIndex := int32(0); hitIndex < numHits; hitIndex++ {
-			baseDamage := sim.Roll(973, 1111) + spellCoeff*spell.SpellPower()
+			baseDamage += spellCoeff * spell.SpellPower()
 			baseDamage *= bounceCoeff
 			result := spell.CalcDamage(sim, curTarget, baseDamage, spell.OutcomeMagicHitAndCrit)
 
-			if canLO && result.Landed() && sim.RandomFloat("CL Lightning Overload") <= lightningOverloadChance {
-				shaman.ChainLightningLOs[hitIndex].Cast(sim, curTarget)
+			if canOverload && result.Landed() && sim.RandomFloat("Chain Lightning Elemental Overload") <= overloadChance {
```

Comment:
For Chain Lightning, Overload chance is divided by 3

Reply:
Thank you. Fixed it.